src/SprykerEco/Zed/Amazonpay/Business/Api/Converter/AbstractResponseParserConverter.php

```diff
@@ -128,16 +136,17 @@ protected function extractHeader(ResponseInterface $responseParser)
     }
 
     /**
-     * @param AmazonpayResponseHeaderTransfer $header
+     * @param \Generated\Shared\Transfer\AmazonpayResponseHeaderTransfer $header
+     *
+     * @return void
      */
     protected function extractErrorFromConstraints(AmazonpayResponseHeaderTransfer $header)
     {
         if ($header->getConstraints()->count() === 0) {
             return;
         }
 
-        /** @var AmazonpayResponseConstraintTransfer $constraint */
-        $constraint = $header->getConstraints()->offsetGet(0);
+        $constraint = $header->getConstraints()[0];
```

Comment:
I would extract to method and give name to it  which would indicate why/what you pick from array first index.